src/bvals/boundary_conditions.cpp

```diff
@@ -78,168 +80,100 @@ TaskStatus ApplyBoundaryConditionsOnCoarseOrFine(std::shared_ptr<MeshBlockData<R
 
 namespace BoundaryFunction {
 
-// TODO(JMM): These are all awfully similar. There's gotta be a way to
-// save some code here.
-void OutflowInnerX1(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
+template <CoordinateDirection DIR, bool INNER, bool REFLECT>
```

Comment:
For the sake of readability, how about using enum classes instead of the `bool`. Given that it's all compile time logic and bool/enum/int typically have the same size, it should also not affect performance.

Reply:
Good idea. I'll make that change